platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -607,6 +607,7 @@ public class SwirldsPlatform implements Platform, Startable {
                 currentAddressBook,
                 selfId,
                 appVersion,
+                initialState.getState().getPlatformState().getPlatformData().getEpochHash(),
```

Comment:
When we do this, we should use the epoch hash defined by the recovery manager if it is not null. If the emergency recovery manager has a null hash, then we should use the epoch hash in the initial state.

During emergency recovery, the epoch hash of the initial state is never correct.

- For the node(s) with the golden state, the hash of the state root matches the epoch hash. But the epoch hash of that state will not because a self referential hash is cryptographically hard to derive.
- For nodes without the golden state, the initial state will either be a genesis state or it will be from the previous epoch.